kafka_consumer_freshness_tracker/src/test/java/com/tesla/data/consumer/freshness/ConsumerFreshnessTest.java

```diff
@@ -272,6 +277,137 @@ public void testBurrowMissingConsumerGroupPartitionEndOffsetMarksMissing() throw
+    });
+  }
+
+  @Test
+  public void testNormalModeConfigurationIsInvalidContainsUnknownServer() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "cluster1";
+    when(burrow.getClusterBootstrapServers(clusterName))
+      .thenReturn(Arrays.asList(
+            "kafka01.example.com:10251", "kafka02.example.com:10251", "kafka03.example.com:10251")
+          );
+
+    Map<String, Object> conf = mockConfForCluster(
+        clusterName,
+        "kafka01.example.com:10251", "kafka02.example.com:10251", "kafka03.example.com:10251",
+            "kafka04.example.com:10251"
+        );
+
+    withExecutor(executor -> {
+      ConsumerFreshness freshness = new ConsumerFreshness();
+
+      // A runtime exception should occur when the burrow call returns 404
+      thrown.expect(RuntimeException.class);
+      freshness.setupWithBurrow(conf, burrow);
+    });
+  }
+
+  @Test
+  public void testStrictModeConfigurationIsValidServersMatchBurrow() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "cluster1";
+    when(burrow.getClusterBootstrapServers(clusterName))
+      .thenReturn(Arrays.asList("kafka01.example.com:10251", "kafka02.example.com:10251",
+              "kafka03.example.com:10251")
+          );
+
+    Map<String, Object> conf = mockConfForCluster(
+        clusterName,
+            "kafka01.example.com:10251", "kafka02.example.com:10251", "kafka03.example.com:10251"
+        );
+
+    withExecutor(executor -> {
+      ConsumerFreshness freshness = new ConsumerFreshness();
+      freshness.strict = true;
+
+      // This exception should be thrown when the consumers are attempted to be
+      // connected after validation succeeds
+      thrown.expect(KafkaException.class);
+      thrown.expectMessage("Failed to construct kafka consumer");
+
+      freshness.setupWithBurrow(conf, burrow);
+    });
+  }
+
+  @Test
+  public void testStrictModeConfigurationIsInvalidMissingServer() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "cluster1";
+    when(burrow.getClusterBootstrapServers(clusterName))
+      .thenReturn(Arrays.asList("kafka01.example.com:10251", "kafka02.example.com:10251",
+              "kafka03.example.com:10251", "kafka04.example.com:10251")
+            );
+
+    Map<String, Object> conf = mockConfForCluster(
+        clusterName,
+        "kafka01.example.com:10251", "kafka02.example.com:10251", "kafka03.example.com:10251"
+        );
+
+    withExecutor(executor -> {
+      ConsumerFreshness freshness = new ConsumerFreshness();
+      freshness.strict = true;
+      // A runtime exception should occur when the burrow call returns 404
+      thrown.expect(RuntimeException.class);
+      freshness.setupWithBurrow(conf, burrow);
+    });
+  }
+
+  Map<String, Object> mockConfForCluster(String name, String... bootstrapServers) {
+    Map<String, Object> conf = new HashMap<String, Object>();
```

Comment:
There should be testing specifically for Burrow handling of the REST response

Reply:
added a test